### PR TITLE
Namespace permission

### PR DESCRIPTION
**Pull request by robertgraeff**
**What this PR does / why we need it**:

This pull request adds a ClusterRole and a ClusterRoleBinding, so that a user of the landscaper can list the namespaces of the resource cluster.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user
NONE
```

### CODE REVIEW

File: pkg/controllers/subjectsync/reconcile_test.go [hunk 1]
```diff
@@ -174,11 +174,15 @@ var _ = Describe("Reconcile", func() {
 
 		updatedLsUserRoleBinding := rbacv1.RoleBinding{}
 		Expect(testenv.Client.Get(ctx, types.NamespacedName{Name: subjectsync.LS_USER_ROLE_BINDING_IN_NAMESPACE, Namespace: lsUserNamespace}, &updatedLsUserRoleBinding)).To(Succeed())
-		Expect(len(updatedLsUserRoleBinding.Subjects)).To(Equal(0))
+		Expect(len(updatedLsUserRoleBinding.Subjects)).To(Equal(1))
```
Comment (enrico-kaack-comp) [inline pkg/controllers/subjectsync/reconcile_test.go:177]: should this be 1 now?

**Review by enrico-kaack-comp — APPROVED** [review]

File: pkg/controllers/namespaceregistration/reconcile_test.go [hunk 2]
```diff
@@ -56,7 +56,7 @@ var _ = Describe("Reconcile", func() {
 		state, err = testenv.InitResources(ctx, "./testdata/reconcile/test1")
 		Expect(err).ToNot(HaveOccurred())
 
-		namespaceRegistration := state.GetNamespaceRegistration("test-namespace-1")
+		namespaceRegistration := state.GetNamespaceRegistration("cu-test-namespace-1")
```
Comment (achimweigel) [inline pkg/controllers/namespaceregistration/reconcile_test.go:59]: Should't we use our constant for `cu-` here?

File: pkg/controllers/namespaceregistration/reconcile_test.go [hunk 3]
```diff
@@ -70,7 +70,7 @@ var _ = Describe("Reconcile", func() {
...
```
Comment (achimweigel) [inline pkg/controllers/namespaceregistration/reconcile_test.go:73]: same as above

File: pkg/controllers/namespaceregistration/reconcile_test.go [hunk 4]
```diff
@@ -108,7 +108,7 @@ var _ = Describe("Reconcile", func() {
 		state, err = testenv.InitResources(ctx, "./testdata/reconcile/test2")
 		Expect(err).ToNot(HaveOccurred())
 
-		namespaceRegistration := state.GetNamespaceRegistration("test-namespace-2")
+		namespaceRegistration := state.GetNamespaceRegistration("cu-test-namespace-2")
```
Comment (achimweigel) [inline pkg/controllers/namespaceregistration/reconcile_test.go:111]: same as above

File: pkg/controllers/subjectsync/reconcile_test.go [hunk 5]
```diff
@@ -127,7 +127,7 @@ var _ = Describe("Reconcile", func() {
 		state, err = testenv.InitResources(ctx, "./testdata/reconcile/test1")
 		Expect(err).ToNot(HaveOccurred())
 
-		subjectlist := state.GetSubjectList("subjectlist")
+		subjectlist := state.GetSubjectList("subjects")
```
Comment (achimweigel) [inline pkg/controllers/subjectsync/reconcile_test.go:130]: Should't we use our constant for `subjects` here?

File: pkg/controllers/subjectsync/reconcile_test.go [hunk 6]
```diff
@@ -164,7 +164,7 @@ var _ = Describe("Reconcile", func() {
 		state, err = testenv.InitResources(ctx, "./testdata/reconcile/test2")
 		Expect(err).ToNot(HaveOccurred())
 
-		subjectlist := state.GetSubjectList("subjectlist")
+		subjectlist := state.GetSubjectList("subjects")
```
Comment (achimweigel) [inline pkg/controllers/subjectsync/reconcile_test.go:167]: same

File: pkg/controllers/subjectsync/reconcile_test.go [hunk 7]
```diff
@@ -187,7 +191,7 @@ var _ = Describe("Reconcile", func() {
 		state, err = testenv.InitResources(ctx, "./testdata/reconcile/test1")
 		Expect(err).ToNot(HaveOccurred())
 
-		subjectlist := state.GetSubjectList("subjectlist")
+		subjectlist := state.GetSubjectList("subjects")
```
Comment (achimweigel) [inline pkg/controllers/subjectsync/reconcile_test.go:194]: same as above

File: pkg/controllers/subjectsync/reconcile_test.go [hunk 8]
```diff
@@ -288,7 +292,7 @@ var _ = Describe("Reconcile", func() {
...
```
Comment (achimweigel) [inline pkg/controllers/subjectsync/reconcile_test.go:295]: same as above

File: pkg/controllers/subjectsync/reconcile_test.go [hunk 9]
```diff
@@ -363,7 +367,7 @@ var _ = Describe("Reconcile", func() {
...
```
Comment (achimweigel) [inline pkg/controllers/subjectsync/reconcile_test.go:370]: same as above

**Review by achimweigel — APPROVED** [review]
/lgtm

on the question if we should use the constants more frequently

File: pkg/controllers/namespaceregistration/reconcile_test.go (hunk 2 above)
Comment (robertgraeff) [reply pkg/controllers/namespaceregistration/reconcile_test.go:59]: done

File: pkg/controllers/namespaceregistration/reconcile_test.go (hunk 4 above)
Comment (robertgraeff) [reply pkg/controllers/namespaceregistration/reconcile_test.go:111]: done

File: pkg/controllers/namespaceregistration/reconcile_test.go (hunk 3 above)
Comment (robertgraeff) [reply pkg/controllers/namespaceregistration/reconcile_test.go:73]: done

**Pull request merged**